vyper/venom/passes/memmerging.py

```diff
@@ -182,15 +189,28 @@ def _barrier():
 
         self._optimize_copy(bb, intervals, copy_inst)
 
+    def _add_zero_interval(self, intervals: list[_Interval], new_inter: _Interval) -> bool:
+        index = bisect_left(intervals, new_inter)
+        intervals.insert(index, new_inter)
+
+        i = max(index - 1, 0)
+        while i < min(index + 1, len(intervals) - 1):
+            merged = intervals[i].inclusive_merge(intervals[i + 1])
+            if merged:
+                del intervals[i + 1]
+            else:
+                i += 1
+
+        return True
+
     # optimize memzeroing operations
     def _optimize_memzero(self, bb: IRBasicBlock, intervals: list[_Interval]):
         for interval in intervals:
             inst = interval.insts[-1]
             if interval.length == 32 and inst.opcode == "calldatacopy":
-                dst = inst.operands[2]
                 inst.opcode = "mstore"
-                inst.operands = [IRLiteral(0), dst]
-            elif interval.length <= 32:
+                inst.operands = [IRLiteral(0), IRLiteral(interval.dst_start)]
+            elif interval.length == 32:
```

Comment:
i guess now that we use inclusive_merge here, this could be a case with multiple `mstore 0` instructions?

Reply:
actually, should we fuse these two branches?
```
    if interval.length == 32:
        inst.opcode = "mstore"
        inst.operands = ...
```